src/applications/bmqstoragetool/m_bmqstoragetool_testutils.h

```diff
@@ -102,8 +102,8 @@ class JournalFile {
     // Number of records.
     MappedFileDescriptor d_mfd;
     // Mapped file descriptor.
-    char* d_mem_p;
-    // Pointer to allocated memory for journal file.
+    bsl::vector<char> d_mem;
```

Comment:
Vector is fine (exception safe!).  I'm surprised there isn't a clear BDE component for this use case, but I couldn't find one with a quick search.

Reply:
I believe allocating data buffers via vector is a common practice for safe (not a joke, but still sounds funny) C++ programming, just as using smart pointers and other RAII things

Reply:
From the past days, we used `levmar` lib for optimizations in C++ code, with API like this: 
```
int dlevmar_dif(
void (*func)(double *p, double *hx, int m, int n, void *adata)
```
And we were really happy to use vector buffers on top of this